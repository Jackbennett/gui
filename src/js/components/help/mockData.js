export const helpProps = {
  docsVersion: '',
  findLocalIpAddress: jest.fn,
  isHosted: false,
  isEnterprise: false,
  menderDebPackageVersion: 'master',
  menderVersion: 'master',
  menderArtifactVersion: 'master',
  org: {
    tenant_token: ''
  }
};
