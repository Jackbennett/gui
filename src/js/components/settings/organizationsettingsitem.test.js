import React from 'react';
import renderer from 'react-test-renderer';
import OrganizationSettingsItem from './organizationsettingsitem';
import { undefineds } from '../../../../tests/mockData';

describe('OrganizationSettingsItem Component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <OrganizationSettingsItem
          title="Current plan"
          content={{
            action: { title: 'Compare product plans', internal: false, target: 'https://mender.io/plans/pricing' },
            description: 'Trial'
          }}
          notification="upgrade now!"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
    expect(JSON.stringify(tree)).toEqual(expect.not.stringMatching(undefineds));
  });
});
