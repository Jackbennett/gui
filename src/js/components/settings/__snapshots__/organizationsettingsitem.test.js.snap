// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrganizationSettingsItem Component renders correctly 1`] = `
<li
  className="MuiListItem-root flexbox column margin-top-small MuiListItem-gutters Mui-disabled MuiListItem-alignItemsFlexStart"
  disabled={true}
  style={
    Object {
      "maxWidth": 500,
    }
  }
>
  <div
    className="MuiListItemText-root MuiListItemText-multiline"
    style={
      Object {
        "display": "grid",
        "marginBottom": 8,
        "width": "100%",
      }
    }
  >
    <span
      className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
    >
      Current plan
    </span>
    <div
      className="MuiTypography-root MuiListItemText-secondary two-columns MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
      style={
        Object {
          "display": "grid",
          "marginBottom": 8,
          "width": "100%",
        }
      }
    >
      <div>
        Trial
      </div>
      <a
        className="align-right"
        href="https://mender.io/plans/pricing"
        target="_blank"
      >
        Compare product plans
      </a>
    </div>
  </div>
  <hr
    className="MuiDivider-root"
    style={
      Object {
        "marginBottom": 8,
        "marginLeft": -16,
        "width": 500,
      }
    }
  />
  upgrade now!
</li>
`;
